aim/storage/query.py

```diff
@@ -151,7 +151,7 @@ def match(
         metric_name: str = None
     ) -> bool:
 
-        context_proxy = AimObjectProxy(lambda: context.to_dict())
+        context_proxy = AimObjectProxy(lambda: context.to_dict() if context else {})
```

Comment:
do we need this though?  this won't throw an exception here, and in the `_check` method the exception will be caught and logged

Reply:
Need to double-check; had a test failing because of this.

Reply:
@mihran113 checked and fixed. The issue was with the logging in `except` block.